src/components/panel/panel.js

```diff
@@ -0,0 +1,284 @@
+ *     possible values: start, center, end, justify. This is only necessary if you do
+ *     not want the panel to intelligently determine location based on screen edges.
+ *       - `top`:
+ *         The panel is aligned to the top of the originating element.
+ *       - `bottom`:
+ *         The panel is aligned to the bottom of the originating element.
+ *   - `overlap` - `{boolean=}`: Whether the panel should overlap the originating
+ *     element. This value is true by default.
+ *   - `center` - `{boolean=}`: If true, centers the panel in the viewport, both
+ *     horizontally and vertically. If this option is set, `alignX` and `alignY` will
+ *     be ignored.
+ *   - `animation` - `{string=}`: The classname for the type of animation. There are
+ *     several built-in animations that you can use:
+ *       `md-animate-fly`: The panel flies in and out from the specified elements.
+ *       `md-animate-scale`: The panels scales in and out.
+ *       `md-animate-none`: No animations.
+ *   - `targetEvent` - `{DOMClickEvent=}`: A click's event object. When passed in as
+ *     an option, the location of the click will be used as the starting point for the
+ *     opening animation of the panel.
+ *   - `openFrom` - `{(string|Element|Object)=}`: The query selector, DOM element,
+ *     or the Rect object that is used to determine the bounds (top, left, height,
+ *     width) from which the panel will originate.
+ *   - `closeTo` - `{(string|Element|Object)=}`: The query selector, DOM element, or
+ *     the Rect object that is used to determine the bounds (top, left, height, width)
+ *     from which the panel will close to.
+ *   - `hasBackdrop` - `{boolean=}`: Whether there should be an opaque backdrop behind
+ *     the panel. Defaults to false.
+ *   - `escapeToClose` - `{boolean=}`: Whether the user can press escape to close the
+ *     panel. Defaults to false.
+ *   - `clickOutsideToClose` - `{boolean=}`: Whether the user can click outside the
+ *     panel to close it. Defaults to false.
+ *   - `disableParentScroll` - `{boolean=}`: Whether the user can scroll the page
+ *     behind the panel. Defaults to false.
+ *   - `fullScreen` - `{boolean=}`: Whether the panel should be full screen. Applies
+ *     the class `.md-panel-fullscreen` to the panel on open. Defaults to false.
+ *   - `trapFocus` - `{boolean=}`: Whether focus should be trapped within the panel.
+ *     If `trapFocus` is true, the user will not be able to interact with the rest of
+ *     the page until the panel is dismissed. Defaults to false.
+ *   - `focusOnOpen` - `{boolean=}`: An option to override focus behavior on open.
+ *     Only disable if focusing some other way, as focus management is required for
+ *     panels to be accessible. Defaults to true.
+ *
+ * @returns {$mdPanelRef} panelRef
+ */
+
+
+
+/**
+ * A service that is used for controlling/displaying panels on the screen.
+ *
+ * @param $rootElement
```

Comment:
Oh awesome! I actually had them and then removed them when I noticed they were rarely included in existing code.

@jelbourn For nullability, the indicators were removed when I ran the demo site, so I didn't include them in the actual code. Is this something we can still use? I think it's helpful to have them in the types, but it's also strange to me to include them only in the developer docs, not the public ones.